spec.html

```diff
@@ -658,6 +658,7 @@
         1. Let _constructorParent_ be the intrinsic object %FunctionPrototype%.
       1. Else,
         1. Set the running execution context's LexicalEnvironment to _classScope_.
+        1. <ins>NOTE: The running execution context's PrivateNameEnvironment is _outerPrivateEnvironment_ when evaluating |ClassHeritage|.</ins>
```

Comment:
Why is this necessary? Should we have more notes explaining the state of various environments, or is this enough?

Reply:
The note in https://tc39.github.io/proposal-class-fields/#sec-execution-contexts gives the impression that implementers could use a single environment for `classScope` and `classPrivateEnvironment`, but during evaluation of `ClassHeritage` the lexical environment is `classScope`, but the private name environment is still `outerPrivateEnvironment`. Which means it's actually not possible to use a single environment for `classScope` and `classPrivateEnvironment`. I think it's worth pointing this out here, isn't it?

Reply:
Do you think we should evaluate the heritage in the private name environment created in the class, then?

Reply:
Not necessarily. I'd probably leave it as is, and only if there's pushback from implementers we could consider pushing the private name environment before evaluating the class heritage expression.

Reply:
cc @gsathya @xanplz